stl/inc/chrono

```diff
@@ -5177,184 +5217,1230 @@ namespace chrono {
+                {'m', _O_mod},
+                {'M', _O_mod},
+                {'S', _O_mod},
+                {'u', _O_mod},
+                {'U', _O_mod},
+                {'V', _O_mod},
+                {'w', _O_mod},
+                {'W', _O_mod},
+                {'x', _E_mod},
+                {'X', _E_mod},
+                {'y', _EO_mod},
+                {'Y', _E_mod},
+                {'z', _EO_mod},
+            };
+
+            const _Allowed_bit _Mod = _Modifier == 'E' ? _E_mod : _O_mod;
+
+            if (auto _It = _RANGES find(_Table, _Type, &_Table_entry::_Type); _It != _STD end(_Table)) {
+                if (_It->_Allowed & _Mod) {
+                    return;
+                }
             }
+
+            _THROW(format_error("Incompatible modifier for type"));
         }
 
-        // chrono-spec
-        while (_Begin != _End && *_Begin != '}') {
-            if (*_Begin == '%') { // conversion-spec
-                if (_Begin + 1 == _End) {
-                    _THROW(format_error("Invalid format string - missing type after %"));
+        template <class _Ty>
+        _NODISCARD static constexpr bool _Is_valid_type(const char _Type) noexcept {
+            if constexpr (_Is_specialization_v<_Ty, duration>) {
+                return _Type == 'j' || _Type == 'q' || _Type == 'Q' || _Is_valid_type<hh_mm_ss<seconds>>(_Type);
+            } else if constexpr (is_same_v<_Ty, day>) {
+                return _Type == 'd' || _Type == 'e';
+            } else if constexpr (is_same_v<_Ty, month>) {
+                return _Type == 'b' || _Type == 'B' || _Type == 'h' || _Type == 'm';
+            } else if constexpr (is_same_v<_Ty, year>) {
+                return _Type == 'Y' || _Type == 'y' || _Type == 'C';
+            } else if constexpr (_Is_any_of_v<_Ty, weekday, weekday_indexed, weekday_last>) {
+                return _Type == 'a' || _Type == 'A' || _Type == 'u' || _Type == 'w';
+            } else if constexpr (_Is_any_of_v<_Ty, month_day, month_day_last>) {
+                return _Type == 'j' || _Is_valid_type<month>(_Type) || _Is_valid_type<day>(_Type);
+            } else if constexpr (_Is_any_of_v<_Ty, month_weekday, month_weekday_last>) {
+                return _Is_valid_type<month>(_Type) || _Is_valid_type<weekday>(_Type);
+            } else if constexpr (is_same_v<_Ty, year_month>) {
+                return _Type == 'g' || _Type == 'G' || _Is_valid_type<year>(_Type) || _Is_valid_type<month>(_Type);
+            } else if constexpr (_Is_any_of_v<_Ty, year_month_day, year_month_day_last, year_month_weekday,
+                                     year_month_weekday_last>) {
+                return _Type == 'D' || _Type == 'F' || _Type == 'g' || _Type == 'G' || _Type == 'j' || _Type == 'U'
+                    || _Type == 'V' || _Type == 'W' || _Is_valid_type<year>(_Type) || _Is_valid_type<month>(_Type)
+                    || _Is_valid_type<day>(_Type) || _Is_valid_type<weekday>(_Type);

```

Comment:
For compound types, we need to rethink how we constrain printing their fields. For `aAuw` we check explicitly if the weekday is ok, even if the year is invalid, but not for other types (such as `b` for `month_day`).

Reply:
Created #1871 for this.

Reply:
Thanks - merged to `chronat2` and applied to this PR.